### PR TITLE
packaging: fix build failure on bionic and simplify rules

### PR DESCRIPTION
**Pull request by mvo5**
For go-1.13 we need to ensure that we have a GOCACHE directory set.
If this is not set then the HOME directory is used. But in an sbuild
environment this is set to /sbuild-nonexistent so the build fails.

The reason we see this on 18.04 now is that we switched to the new
go-1.13 there but the dh-golang tooling is still assuming the 18.04
go-1.10.

This commit fixes this by setting one global GOCACHE.

See https://launchpad.net/~snappy-dev/+archive/ubuntu/edge/+packages?field.name_filter=&field.status_filter=published&field.series_filter=bionic for the build failures.

### CODE REVIEW

File: packaging/debian-sid/rules [hunk 1]
```diff
@@ -17,6 +17,8 @@ export DH_GOLANG_EXCLUDES=tests
 export DH_GOLANG_GO_GENERATE=1
 
 export PATH:=${PATH}:${CURDIR}
+# GOCACHE is needed by go-1.13
+export GOCACHE:=/tmp/go-build
```
Comment (anonymouse64) [inline packaging/debian-sid/rules:21]: do we need this for 14.04 too?

Comment (bboozzoo) [reply packaging/debian-sid/rules:21]: IIRC this was only introduced in 1.12 or 1.13, so we probably do not need that on Trusty if building with go from the repositories.

**Review by bboozzoo — APPROVED** [review]
LGTM

File: packaging/debian-sid/rules (hunk 1 above)
Comment (mvo5) [reply packaging/debian-sid/rules:21]: We also have this in packaging/ubuntu-14.04/rules

Comment (anonymouse64) [reply packaging/debian-sid/rules:21]: ah right good point it's already there

**Review by anonymouse64 — APPROVED** [review]
lgtm

**Pull request merged**